### DISCUSSION TITLE
Balancing open-source and a full-time job (Sniffnet’s 2nd anniversary)

### DISCUSSION THREAD

**Discussion by GyulyVGC** [GyulyVGC/sniffnet#574] [Announcements]
_Woah_. 
Another year has flown away.

It seems like yesterday that I was writing down the [post](https://github.com/GyulyVGC/sniffnet/discussions/329) to celebrate Sniffnet’s first anniversary.
On that day, I was coming from a period in which I totally dedicated myself to the app development. 
Like 12 hours per day, and that was cool.

<p>
<picture>
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</picture>
</p>

### The crossroads 🔀

However, in the long run, **Sniffnet alone wasn’t enough to pay the bills** — even a part-time job is way more profitable.
For this reason, I decided to open myself to new opportunities, and last September **I was hired as a full-time Rust developer** on a cybersecurity product.
I was lucky enough to get hired for a _remote_ position, allowing me to code from the comfort of my room, where I’m the most inspired and productive.
Moreover, the fact that I’m still using Rust on a daily basis makes it awesome — _I really can’t complain_.

_Well… then what?_
Being busy on other software projects 40 hours per week obviously doesn’t allow me to develop Sniffnet at the pace I used to.
At first, I thought I was at a _**crossroads**_: 
- Sniffnet on one side
- a full-time job on the other side

<div align="center">
<picture>
<img alt="" title="The crossroads" src="https://github.com/user-attachments/assets/2c6f40d0-7690-49c8-8aeb-c4046c23aa0c" width="50%"/>
</picture>
</div>

It’s a tough decision.
In such circumstances, taking the full-time job road seemed the wisest and most reasonable thing to do.
But what if I tell you that it doesn’t have to be a _decision_?
I mean, it’s not physically a crossroads, right?

<p>
<picture>
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</picture>
</p>

### The puzzle 🧩

Something I understood during the past year is that every single activity we perform brings added value to us as professionals, but most importantly as _thinkers_ and as _people_.
Instead of seeing it as a crossroads, I now like to see it as _**two pieces of the same puzzle**_: they can live on their own, but together they can form a bigger and more complete picture.

<div align="center">
<picture>
<img alt="" title="The puzzle" src="https://github.com/user-attachments/assets/4ac948ac-7f21-4d86-9f5f-d41ed2247f05" width="50%"/>
</picture>
</div>

Every working day, I notice how helpful are the skills that I built through thousands of hours maintaining Sniffnet.
Without the experience gained from open-source development, I wouldn’t have even been hired for my current position in the first place.
At the same time, digging further into Rust at work is enabling me to master and deeply understand aspects that I can now better apply to Sniffnet itself (generics, trait objects, and lifetimes, just to name a few).

Sure, I have less time for Sniffnet now.
But I’m no longer sure that it’s a real loss — neither for me nor for the project itself.
I come back to Sniffnet when I feel to — _without any pressure and with a bag of new experiences_ — also thanks to the fact that remote jobs eliminate commute times and have more flexible schedules.

During the past year, a new [major version](https://github.com/GyulyVGC/sniffnet/releases/tag/v1.3.0) was released, as well as a new [minor version](https://github.com/GyulyVGC/sniffnet/releases/tag/v1.3.1) just a few days ago.
The project is in **constant expansion**, and I’m really glad to see that many new users are still adopting Sniffnet.
Just considering the past week, the repository got about 1k more stars, the app was downloaded 10k more times, and Sniffnet is now the most popular GitHub project by the following topics: [#utility](https://github.com/topics/utility), [#pcap](https://github.com/topics/pcap), [#network-monitoring](https://github.com/topics/network-monitoring), [#packet-sniffer](https://github.com/topics/packet-sniffer), and more.

<div align="center">
<picture>
<img alt="" title="Adoption on Arch Linux" src="https://github.com/user-attachments/assets/ff675bf1-2937-4254-984b-ad2e19ae3385" width="52%"/>
</picture>
<picture>
<img alt="" title="GitHub stars" src="https://github.com/user-attachments/assets/796c974e-08d0-4cec-aa55-a5380150d4a3" width="42%"/>
</picture>
<picture>
<img alt="" title="Rust trending repository of the day" src="https://github.com/user-attachments/assets/d692c9d0-a3cb-4849-8cdb-718dd0bad083" width="95%"/>
</picture>
</div>

<p>
<picture>
<img alt="" src="https://github.com/GyulyVGC/sniffnet/blob/main/resources/repository/hr.png?raw=true" width="100%"/>
</picture>
</p>

### Wrap up 🤝

Often, we believe that having a _new priority_ implies _giving up_ on something else.
But it’s probably enough to shift our point of view in order to notice that a change in our habits is an _opportunity_ that can make us grow even further.
It’s never too late to go back and retrace our steps, while an opportunity not seized is something that may turn into a permanent regret.

_**Not all the pieces always fit together.
But if we don’t even try to build the puzzle, we won’t ever be able to grasp the whole picture.**_

**Comment by gyreas** [comment]
Well said! This has been a huge source of inspiration for me. Thanks and I wish you success at your work.

**Reply by GyulyVGC** [reply]
Thank you!

**Comment by AmintaCCCP** [comment]
Congratulations and good luck with the new job.

**Reply by GyulyVGC** [reply]
🙏
